src/test/scala/io/iohk/ethereum/consensus/pow/PoWMinerCoordinatorSpec.scala

```diff
@@ -0,0 +1,192 @@
+package io.iohk.ethereum.consensus.pow
+
+import akka.actor.testkit.typed.LoggingEvent
+import akka.actor.testkit.typed.scaladsl.LoggingTestKit
+import akka.actor.typed.scaladsl.adapter._
+import akka.actor.{ActorRef, ActorSystem, typed}
+import akka.testkit.{TestActor, TestKit, TestProbe}
+import io.iohk.ethereum.blockchain.sync.SyncProtocol.MinedBlock
+import io.iohk.ethereum.consensus.pow.PoWMiningCoordinator._
+import io.iohk.ethereum.domain.{Block, UInt256}
+import io.iohk.ethereum.jsonrpc.EthMiningService.SubmitHashRateResponse
+import io.iohk.ethereum.ommers.OmmersPool
+import io.iohk.ethereum.transactions.PendingTransactionsManager
+import io.iohk.ethereum.{Fixtures, WithActorSystemShutDown}
+import monix.eval.Task
+import org.bouncycastle.util.encoders.Hex
+import org.scalatest.flatspec.AnyFlatSpecLike
+import org.scalatest.matchers.should.Matchers
+
+import scala.concurrent.duration._
+
+class PoWMinerCoordinatorSpec
+// avoid exception "Only adapted classic ActorContext permissible" by using untyped TestKit
+    extends TestKit(ActorSystem("PoWMinerCoordinatorSpec_System"))
+    with AnyFlatSpecLike
+    with WithActorSystemShutDown
+    with Matchers {
+
+  implicit val systemTyped: typed.ActorSystem[Nothing] = system.toTyped
+
+  "PoWMinerCoordinator actor" should "throw exception when starting with other message than StartMining(mode)" in new TestSetup(
+    "FailedCoordinator"
+  ) {
+    LoggingTestKit.error("Supervisor StopSupervisor saw failure: StopMining").expect {
+      coordinator ! StopMining
+    }
+  }
+
+  it should "start recurrent mining when receiving message StartMining(RecurrentMining)" in new TestSetup(
+    "RecurrentMining"
+  ) {
+    setBlockForMining(parentBlock)
+    LoggingTestKit.info("Received message StartMining(RecurrentMining)").expect {
+      coordinator ! StartMining(RecurrentMining)
+    }
+    coordinator ! StopMining
+  }
+
+  it should "start on demand mining when receiving message StartMining(OnDemandMining)" in new TestSetup(
+    "OnDemandMining"
+  ) {
+    LoggingTestKit.info("Received message StartMining(OnDemandMining)").expect {
+      coordinator ! StartMining(OnDemandMining)
+    }
+    coordinator ! StopMining
+  }
+
+  it should "[Recurrent Mining] ProcessMining starts EthashMiner if mineWithKeccak is false" in new TestSetup(
+    "EthashMining"
+  ) {
+    setBlockForMining(parentBlock)
+    LoggingTestKit.info("Spawning an EthashMiner").expect {
+      coordinator ! StartMining(RecurrentMining)
+    }
+
+    coordinator ! StopMining
+  }
+
+  it should "[Recurrent Mining] ProcessMining starts KeccakMiner if mineWithKeccak is true" in new TestSetup(
+    "KeccakMining"
+  ) {
+    override val coordinator = system.spawn(
```

Comment:
wouldn't it make sense to extract the duplication for the `spawn` of the `PoWMiningCoordinator`? eg. move ti to `TestSetup`?